### DISCUSSION TITLE
add header files to vs project generator

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1921] [Q&A]
### 你在什么场景下需要该功能？

现在无论是vs还是vsxmake，自动生成的solution里面都仅有源文件，没有头文件，给头文件的浏览、修改带来很大麻烦。

### 描述可能的解决方案

把头文件也加入solution中。可以自动搜索项目头文件，也可以由用户声明哪些是头文件，还可以类似cmake，支持add_files("*.h")，这些头文件不参与编译但能够输出到最后的solution文件中

### 其他问题

xmake目前在头文件发生变化的时候重新编译的机制是怎样的？xmake.lua中没有任何关于头文件的信息，所以xmake是支持自动检索用到的头文件的吗

**Comment by waruqi** [comment]
不是有 add_headerfiles 么，vs/vsxmake 都支持

**Comment by SirLynix** [comment]
I'm trying.

XMake可以用`add_headerfiles`向Visual Studio项目添加头文件。

它通过编译器选项知道头文件的变化，输出包含的头文件的列表。

**Comment by xq114** [comment]
试了下确实
